monai/fl/client/monai_algo.py

```diff
@@ -55,9 +56,12 @@ def convert_global_weights(global_weights, local_var_dict):
                 weights = torch.reshape(torch.as_tensor(weights), local_var_dict[var_name].shape)
                 # update the local dict
                 local_var_dict[var_name] = weights
+                n_converted += 1
             except Exception as e:
                 raise ValueError(f"Convert weight from {var_name} failed.") from e
-    return local_var_dict
+    if n_converted == 0:
+        raise ValueError(f"No global weights converted! Received weight dict keys are {list(global_weights.keys())}")
```

Comment:
perhaps this could be a warning saying received weight dict but no matched parameter keys?

Reply:
Right. Continuing here would cause an error later when computing weight differences because the local and global weights end up on different devices.

Is there a use case where train() is called but provides empty weights in the `data`? It would be unusual for the standard federated averaging workflow we want to implement here. @habibcomsats , @Warvito, any opinions?

Reply:
Actually, we can support this by moving the weight diff computation to the CPU() for both local and global model. Then just printing the warning should be sufficient.

Reply:
@wyli, I made some changes that move all returned weights to CPU, allowing also to compute differences if non have been converted. Now, only a warning is printed in that case.